playbooks/robusta_playbooks/api_service.py

```diff
@@ -0,0 +1,139 @@
+import json
+import logging
+from typing import List, Optional
+
+import yaml
+from kubernetes.client import ApiregistrationV1Api, V1APIServiceList
+from pydantic import BaseModel
+from robusta.api import BaseBlock, FileBlock, MarkdownBlock, PrometheusKubernetesAlert, action
+
+
+class ApiServiceConditions(BaseModel):
+    message: Optional[str]
+    reason: Optional[str]
+    status: Optional[str]
+    type: Optional[str]
+
+
+class ApiService(BaseModel):
+    name: str
+    service_name: Optional[str]
+    service_namespace: Optional[str]
+    conditions: List[ApiServiceConditions]
+    full_json: dict
+
+    def get_json(self):
+        return self.full_json
+
+    def get_yaml(self):
+        # remove noisy fields
+        if self.full_json.get("metadata", {}).get("managedFields", {}):
+            del self.full_json["metadata"]["managedFields"]
+        last_applied = "kubectl.kubernetes.io/last-applied-configuration"
+        if self.full_json.get("metadata", {}).get("annotations", {}).get(last_applied):
+            del self.full_json["metadata"]["annotations"][last_applied]
+        return yaml.dump(self.full_json)
+
+    def get_available_condition(self) -> Optional[ApiServiceConditions]:
+        if not self.conditions:
+            return None
+        available_conditions = [condition for condition in self.conditions if condition.type == "Available"]
+        if not available_conditions:
+            return None
+        return available_conditions[0]
+
+    @staticmethod
+    def parse_apiservice(apiservice: dict, name: str):
```

Comment:
I think it will be more concise to move the code to the constructor , if its possible